core/tests/ts-integration/tests/api/web3.test.ts

```diff
@@ -932,6 +934,21 @@ describe('web3 API compatibility tests', () => {
         expect(txFromApi.v! <= 1).toEqual(true);
     });
 
+    // We want to be sure that correct(outer) contract address is return in the transaction receipt,
+    // when there is a contract that initializa another contract in the constructor
+    test('Should check inner-outer contract address in the receipt of the deploy tx', async () => {
+        let innerContractBytecode = contracts.inner.bytecode;
+        let outerContractOverrides = {
+            customData: {
+                factoryDeps: [innerContractBytecode]
+            }
+        };
+        const outerContract = await deployContract(alice, contracts.outer, [1], undefined, outerContractOverrides);
+        let receipt = await outerContract.deployTransaction.wait();
+
+        expect(outerContract.address).toEqual(receipt.contractAddress);
```

Comment:
I'm not sure how address which is returned from `deployContract` is calculated. If it's taken from `receipt.contractAddress` then the assert doesn't check what we want to check. Let's add another `expect` that checks that bytecode hash of `receipt.contractAddress` is indeed bytecode hash of outer

Reply:
Hmmm....I thought once again, and I do not believe that checking bytecode hashes will increase test robustness.
I have decided to check receipt contract address with precalculated address throw `utils.createAddress`

Reply:
Both works